### PR TITLE
[flutter_tools] remove global usage of flutterUsage from gradle

### PR DESCRIPTION
**Pull request by jonahwilliams**
Also moves project locating test to new file.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/android/gradle_find_bundle_test.dart [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+  bundleDirectory
+    .childDirectory(directoryName)
+    .childFile(fileName)
+    .createSync();
+  return project;
+}
+
+class MockAndroidProject extends Mock implements AndroidProject {}
+class MockFlutterProject extends Mock implements FlutterProject {}
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/android/gradle_find_bundle_test.dart:364]: Are these dead in gradle_test.dart?

**Comment by jmagman** [comment]
Failing `android_gradle_builder_test` tests https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8854924732232723264/+/steps/run_test.dart_for_tool_tests_shard_and_subshard_general/0/stdout

**Comment by jonahwilliams** [comment]
Woops, fixed!

File: packages/flutter_tools/test/general.shard/android/gradle_find_bundle_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/android/gradle_find_bundle_test.dart:366]: They are not unfortunately

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**